cfripper/rules/GenericWildcardPrincipalRule.py

```diff
@@ -64,7 +62,7 @@ def check_for_wildcards(self, logical_id: str, resource: PolicyDocument):
                     if account_id_match:
                         self.validate_account_id(logical_id, account_id_match.group(1))
 
-                    if statement.Condition is not None:
+                    if statement.Condition and statement.Condition.dict():
```

Comment:
What is the check for `statement.Condition.dict()` doing exactly? :)

Reply:
`statement.Condition` is an object. So the object can contain or not any information and still exist. So the condition would fail. To fix it, we convert the object to a dict, and we check if dict is empty or not. Ideally,  pycfmodel in the future will have a custom `__bool__` function for these cases.